src/protections/MessageIsMedia.ts

```diff
@@ -35,10 +35,11 @@ export class MessageIsMedia extends Protection {
 
     public async handleEvent(mjolnir: Mjolnir, roomId: string, event: any): Promise<any> {
         if (event['type'] === 'm.room.message') {
-            const content = event['content'] || {};
+            let content = event['content'] || {};
+            content = content?.["m.new_content"] ?? content;
```

Comment:
we should probably also check `m.relates_to` to ensure we're actually looking at an edit here. The risk being that someone decides to send partially malformed events to trick Mjolnir into (dis)allowing something which is still rendered to users.

Reply:
So we can assume it's an edit if there is both an `m.new_content` and an `m.relates_to` field?

Reply:
`m.relates_to` can mean [many things](https://spec.matrix.org/v1.11/client-server-api/#relationship-types), but inspecting it should tell us if it's an edit. In practice this means ensuring the `m.relates_to` `rel_type` is `m.replace`, I think. There's a bunch of [validity rules](https://spec.matrix.org/v1.11/client-server-api/#validity-of-replacement-events) we should probably be checking too, but that might still have false negatives if the code isn't perfect (or someone finds a way to send something weird).